### ISSUE TITLE
Confusing behavior for base code in the GST-algorithm (potential bug)

### ISSUE THREAD

**Issue by tsaglam** [jplag/JPlag#171]
For base code submissions, there are certain assumptions in the [greedy string tiling algorithm](https://github.com/jplag/jplag/blob/master/jplag/src/main/java/jplag/GreedyStringTiling.java).
E.g. the second submission is treated differently than the first one:
```java
if (first.hash_length != minimalTokenMatch) {
       createHashes(first, minimalTokenMatch, withBaseCode); // don't make table if not a base code comparison
}
if (second.hash_length != minimalTokenMatch || second.tokenHashes == null) {
        createHashes(second, minimalTokenMatch, true);
}
```
The second submission is also initially the base code one:
```java
JPlagComparison baseCodeMatch = greedyStringTiling.compareWithBaseCode(currentSubmission, baseCodeSubmission);
```
However, as for normal comparisons, the submissions might be swapped (the base code should often be the smaller one, right?):
```java
if (firstSubmission.tokenList.size() > secondSubmission.tokenList.size()) {
    smallerSubmission = secondSubmission;
    largerSubmission = firstSubmission;
}
```
I am not sure if that is the intended behavior for the base code submissions. We should test that and see if we can see a difference in the quality of the base code recognition in the reports. Note that this issue is not newly introduced, but also persists in the [legacy version](https://github.com/jplag/jplag/blob/legacy/jplag/src/main/java/jplag/GSTiling.java).

**Comment by Alberth289346** [comment]
Don't know if you are aware, but #49 has a "Fix to an apparent bug with submission flipping" commit 3fdc7472a97e654def3702267e62fe9eade53424

**Comment by tsaglam** [comment]
This is already fixed in the master branch, see #150. Thanks for bringing it up though!
